File: android/app/src/main/java/com/noor/app/widget/BootReceiver.java
package com.noor.app.widget;

import android.appwidget.AppWidgetManager;
import android.content.BroadcastReceiver;
import android.content.ComponentName;
import android.content.Context;
import android.content.Intent;

public class BootReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        String action = intent.getAction();
        if (Intent.ACTION_BOOT_COMPLETED.equals(action) ||
            "android.intent.action.QUICKBOOT_POWERON".equals(action)) {
            AppWidgetManager awm = AppWidgetManager.getInstance(context);
            int[] ids = awm.getAppWidgetIds(new ComponentName(context, PrayerWidget.class));
            if (ids.length > 0) {
                PrayerWidgetService.start(context);
            }
        }
    }
}

File: artifacts/noor/android-widget/java/widget/BootReceiver.kt
package YOUR_PACKAGE_NAME.widget

import android.appwidget.AppWidgetManager
import android.content.BroadcastReceiver
import android.content.ComponentName
import android.content.Context
import android.content.Intent

/**
 * Restarts the PrayerWidgetService after device reboot.
 * Requires RECEIVE_BOOT_COMPLETED permission in AndroidManifest.xml
 *
 * REPLACE "YOUR_PACKAGE_NAME" with your actual package name.
 */
class BootReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val action = intent.action ?: return
        if (action == Intent.ACTION_BOOT_COMPLETED ||
            action == "android.intent.action.QUICKBOOT_POWERON"
        ) {
            val awm = AppWidgetManager.getInstance(context)
            val ids = awm.getAppWidgetIds(
                ComponentName(context, PrayerWidget::class.java)
            )
            if (ids.isNotEmpty()) {
                PrayerWidgetService.start(context)
            }
        }
    }
}
